### PR TITLE
Tools: improve sitl-on-hw copter handling

### PR DESCRIPTION
**Pull request by khancyr**
Add a bit more help on SITL on HW for copter frame selection ! 
Tested working with octa-quad

### CODE REVIEW

File: Tools/scripts/sitl-on-hardware/sitl-on-hw.py [hunk 1]
```diff
@@ -22,33 +45,46 @@
...
     print("Running (%s)" % " ".join(cmd_list))
     retcode = subprocess.call(cmd_list)
     if retcode != 0:
-       print("FAILED: %s" % (' '.join(cmd_list)))
-       global extra_hwdef
-       if extra_hwdef is not None:
-           extra_hwdef.close()
-           os.unlink(extra_hwdef.name)
-       sys.exit(1)
+        print("FAILED: %s" % (' '.join(cmd_list)))
```
Comment (peterbarker) [inline Tools/scripts/sitl-on-hardware/sitl-on-hw.py:54]: On commit for flake8-clean, one for the changes would have been good!

File: Tools/scripts/sitl-on-hardware/sitl-on-hw.py [hunk 2]
```diff
@@ -63,14 +99,41 @@ def sohw_path(fname):
...
+            "dodeca-hexa": "AP_MOTORS_FRAME_DODECAHEXA_ENABLED",
+            "octa": "AP_MOTORS_FRAME_OCTA_ENABLED",
+            "octa-dji": "AP_MOTORS_FRAME_OCTA_ENABLED",
+            "octa-cwx": "AP_MOTORS_FRAME_OCTA_ENABLED",
+            "octa-quad": "AP_MOTORS_FRAME_OCTAQUAD_ENABLED",
+            "octa-quad-cwx": "AP_MOTORS_FRAME_OCTAQUAD_ENABLED",
+            "y6": "AP_MOTORS_FRAME_Y6_ENABLED"
+        }
+        for frame, define in frame_defines.items():
+            if args.frame == frame:
```
Comment (peterbarker) [inline Tools/scripts/sitl-on-hardware/sitl-on-hw.py:133]: We're missing a frame-not-found error

**Review by peterbarker — CHANGES_REQUESTED** [review]
Nice!

Please also mark the file as `AP_FLAKE8_CLEAN`

**Comment by khancyr** [comment]
@peterbarker updated: split styles and functionals changes, added error handling

File: Tools/scripts/sitl-on-hardware/sitl-on-hw.py [hunk 3]
```diff
@@ -82,12 +152,14 @@ def sohw_path(fname):
 configure_args.extend(unknown_args)
 run_program(configure_args)
 
-build_cmd = ["./waf", args.vehicle]
+# Remove "Ardu", "Heli", and "Antenna"
+waf_vehicle = args.vehicle.lower().replace("ardu", "").replace("heli", "").replace("antenna", "")
```
Comment (peterbarker) [inline Tools/scripts/sitl-on-hardware/sitl-on-hw.py:156]: Uh..... how can we remove "Heli" here?  If someone says `./waf helicopter`, they don't want `waf copter`, surely?

Comment (peterbarker) [reply Tools/scripts/sitl-on-hardware/sitl-on-hw.py:156]: .... also, does `./waf tracker` actually work?  Doesn't that also need to be `./waf antennatracker`?

Comment (khancyr) [reply Tools/scripts/sitl-on-hardware/sitl-on-hw.py:156]: yep, right, I tottally overlook this . I will correct that

Comment (khancyr) [reply Tools/scripts/sitl-on-hardware/sitl-on-hw.py]: done

**Comment by khancyr** [comment]
@peterbarker update with correct target for waf

**Comment by peterbarker** [comment]
@khancyr please outline what testing you've given this now; the tracker and heli things obviously wouldn't have worked.... no big deal as I don't think people usually put those onto vehicles, but ....

**Comment by khancyr** [comment]
I have tested Copter and build test plane. 
Indeed Heli and tracker are failing on frame handling ... but this part is copied from sim_vehicle.py and it doesn't work there either (try sim_vehicle.py -v Helicopter) 

The waf target handling is tested : 

`In [5]: vehicle_map = {
   ...:     "APMrover2": "Rover",
   ...:     "Copter": "ArduCopter",
   ...:     "Heli": "Helicopter",
   ...:     "Plane": "ArduPlane",
   ...:     "Sub": "ArduSub",
   ...:     "Blimp": "Blimp",
   ...:     "Rover": "Rover",
   ...:     "AntennaTracker": "AntennaTracker",
   ...: }
   ...: # add lower-case equivalents too
   ...: for k in list(vehicle_map.keys()):
   ...:     vehicle_map[k.lower()] = vehicle_map[k]
   ...:

In [6]: vehicle_list =["ArduCopter","Helicopter","Blimp","ArduPlane","Rover","ArduSub","AntennaTracker","sit
   ...: l_periph_universal","APMrover2","Copter","Plane","Sub","Blimp","Rover","apmrover2","copter","plane",
   ...: "sub","blimp","rover"]

In [7]: def get_key_from_value(d, target_value):
   ...:     for key, value in d.items():
   ...:         if value == target_value:
   ...:             return key
   ...:     return None
   ...:

In [8]: for vehicle in vehicle_list:
   ...:     if vehicle in ["APMrover2", "apmrover2"]:
   ...:         vehicle = "Rover"
   ...:     print(f"{vehicle if vehicle in vehicle_map.keys() else get_key_from_value(vehicle_map, vehicle)}
   ...: ")
   ...:
Copter
Heli
Blimp
Plane
Rover
Sub
AntennaTracker
None
Rover
Copter
Plane
Sub
Blimp
Rover
Rover
copter
plane
sub
blimp
rover
`

**Comment by peterbarker** [comment]
Merged, thanks!

**Comment by khancyr** [comment]
will try to think a way to manage better those naming handling accross the script (sim_vehicle, autotest, waf, sit_on_hw)

**Pull request merged**